src/main/java/com/ss/training/lms/controller/LibrarianController.java

```diff
@@ -0,0 +1,163 @@
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+public class LibrarianController {
+
+    @Autowired
+    LibrarianService librarianService;
+
+    @GetMapping(path="/lms/librarian/branches",
+                produces = {
+                    MediaType.APPLICATION_XML_VALUE,
+                    MediaType.APPLICATION_JSON_VALUE
+                })
+    public ResponseEntity<List<LibraryBranch>> getBranches()
+     {
+        HttpStatus status = HttpStatus.OK;
+        List<LibraryBranch> branches = null;
+        try {
+            branches = librarianService.getBranches();
+            if (branches == null){
+                status = HttpStatus.NOT_FOUND;
+            }
+		} catch (final SQLException e) {
+            e.printStackTrace();
+            return new ResponseEntity<List<LibraryBranch>>(branches , HttpStatus.INTERNAL_SERVER_ERROR);
+		}
+        return new ResponseEntity<List<LibraryBranch>>(branches , status);
+     }
+
+
+    @RequestMapping(path="/lms/librarian/books/{search}",
+                    produces = {
+                        MediaType.APPLICATION_XML_VALUE,
+                        MediaType.APPLICATION_JSON_VALUE
+                    })
+    public ResponseEntity<List<Book>> getBooksWithSearch(@PathVariable String search)
+    {
+        List<Book> books = null;;
+        HttpStatus status = HttpStatus.OK;
+
+        try {
+            books = librarianService.getBooksWithSearch(search);
+            if (books == null)
+            {
+                status = HttpStatus.NOT_FOUND;
+            }
+        } catch (SQLException e) {
+            e.printStackTrace();
+            return new ResponseEntity<List<Book>>(books , HttpStatus.INTERNAL_SERVER_ERROR);
+
+        }
+        return new ResponseEntity<List<Book>>(books , status);
+
+    }
+
+    @RequestMapping(path="lms/librarian/branches/{branch}/books/{book}/copies",
+                    produces = {
+                        MediaType.APPLICATION_XML_VALUE,
+                        MediaType.APPLICATION_JSON_VALUE
+                    })
+    public ResponseEntity<BookCopies> getAnEntryOfBookCopies(@PathVariable int branch, @PathVariable int book)
+    {
+        BookCopies bookCopies = null;
+        HttpStatus status = HttpStatus.OK;
+
+        try {
+            bookCopies = librarianService.getAnEntryOfBookCopies(branch, book);
+            if (bookCopies == null)
+            {
+                status = HttpStatus.NOT_FOUND;
+            }
+        } catch (SQLException e) {
+            e.printStackTrace();
+            return new ResponseEntity<BookCopies>(bookCopies , HttpStatus.BAD_REQUEST);
```

Comment:
It looks like you missed this one when changing the Bad Requests to Internal Server Errors.